lib/internal/test_runner/test.js

```diff
@@ -555,7 +558,15 @@ class Test extends AsyncResource {
         }
       });
     }
-    ArrayPrototypePush(this.hooks[name], hook);
+    if (name === 'afterEach') {
+      // afterEach hooks for the current test should run in the order that they
+      // are created. However, the current test's afterEach hooks should run
+      // prior to any ancestor afterEach hooks.
+      ArrayPrototypeSplice(this.hooks[name], this.hooks.ownAfterEachCount, 0, hook);
+      this.hooks.ownAfterEachCount++;
```

Comment:
```suggestion
      ArrayPrototypeSplice(this.hooks[name], this.hooks.ownAfterEachCount++, 0, hook);
```

Reply:
I personally find that a little less readable, only because the splice line is already a bit busy.

Reply:
OK, works for me 🙂